xds/internal/xdsclient/dump.go

```diff
@@ -22,22 +22,42 @@ import (
 	"google.golang.org/grpc/xds/internal/xdsclient/xdsresource"
 )
 
+func mergeMaps(maps []map[string]xdsresource.UpdateWithMD) map[string]xdsresource.UpdateWithMD {
+	ret := make(map[string]xdsresource.UpdateWithMD)
+	for _, m := range maps {
+		for k, v := range m {
+			ret[k] = v
+		}
+	}
```

Comment:
Are we guaranteed that keys from different authorities are unique?

Reply:
All the resource names should have the authority in it.

Also see this: https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdsresource/name.go#L28